cloudSupport/src/main/scala/cromwell/cloudsupport/auth/AwsModes.scala

```diff
@@ -0,0 +1,121 @@
+package cromwell.cloudsupport.auth
+
+import cromwell.cloudsupport.auth.AuthMode.OptionLookup
+import software.amazon.awssdk.auth.credentials._
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sts.StsClient
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest
+
+object AwsModes {
+
+  import CredentialValidatorInstances.awsCredentialValidation
+  import CredentialValidatorSyntax._
+
+  case object MockAuthMode extends AwsAuthMode {
+
+    override val _credential: AwsCredentials = AnonymousCredentialsProvider.create.resolveCredentials()
+
+    override val name = "no_auth"
+
+    override def credentials(options: OptionLookup): AwsCredentials = _credential
+
+  }
+
+  object CustomKeyMode
+
+  final case class CustomKeyMode(override val name: String,
+                                 accessKey: String,
+                                 secretKey: String,
+                                 region: Option[String]
+                                ) extends AwsAuthMode {
+
+    override lazy val _credential: AwsCredentials = {
+
+      val storage = Creds.apply(Option(AwsBasicCredentials.create(accessKey, secretKey)), Option(null), region)
+      // Validate credentials synchronously here, without retry.
+      // It's very unlikely to fail as it should not happen more than a few times
+      // (one for the engine and for each backend using it) per Cromwell instance.
+      CredentialValidatorSyntax(storage).validate
+    }
+
+    override def credentials(options: OptionLookup): AwsCredentials = _credential
+  }
+
+  final case class DefaultMode(override val name: String, region: Option[String]) extends AwsAuthMode {
+
+    override lazy val _credential: AwsCredentials = {
+      // The ProfileCredentialsProvider will return your [default]
+      // credential profile by reading from the credentials file located at
+      // (~/.aws/credentials).
+      //
+      // Validate credentials synchronously here, without retry.
+      // It's very unlikely to fail as it should not happen more than a few times
+      // (one for the engine and for each backend using it) per Cromwell instance.
+      Creds.apply(Option(DefaultCredentialsProvider.create.resolveCredentials()), Option(null), region).validate
+    }
```

Comment:
At some cases (for ex. at AssumeRoleMode) laziness used as decision to avoid forward references. But anyway it is unneccessary and much better to move specified fields to the top of the class.